test/test-profiler.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import * as delay from 'delay';
+import * as extend from 'extend';
+import * as nock from 'nock';
+import * as sinon from 'sinon';
+import {instance, mock, when} from 'ts-mockito';
+import * as zlib from 'zlib';
+
+import {ProfilerConfig} from '../src/config';
+import {perftools} from '../src/profile';
+import {Profiler} from '../src/profiler';
+import {TimeProfiler} from '../src/profilers/time-profiler';
+import {base64TestProfile, decodedTestProfile, testProfile} from './profiles-for-tests';
+
+const v8TimeProfiler = require('bindings')('time_profiler');
+
+const fakeCredentials = require('../../test/testdata/gcloud-credentials.json');
+
+const testConfig: ProfilerConfig = {
+  projectId: 'test-projectId',
+  logLevel: 1,
+  serviceContext: {service: 'test-service', version: 'test-version'},
+  instance: 'test-instance',
+  zone: 'test-zone',
+  disableTime: false,
+  disableHeap: false,
+  credentials: fakeCredentials,
+  timeSamplingIntervalMicros: 1000,
+  minTimeBetweenProfilesMillis: 1000 * 60
+};
+
+const API = 'https://cloudprofiler.googleapis.com/v2';
+
+const mockTimeProfiler = mock(TimeProfiler);
+when(mockTimeProfiler.profile(10 * 1000)).thenReturn(new Promise((resolve) => {
+  resolve(testProfile);
+}));
+
+nock.disableNetConnect();
+export function oauth2(): nock.Scope {
```

Comment:
* This needs a better name given that it is supposed to be used directly now. `nockOauth2` would be better.
* This does not need to be exported.